src/decoder_wav.cpp

```diff
@@ -30,24 +30,21 @@ WavDecoder::WavDecoder()
 }
 
 WavDecoder::~WavDecoder() {
-	if (file_ != NULL) {
-		fclose(file_);
-	}
 }
 
-bool WavDecoder::Open(FILE* file) {
-	file_=file;
-	fseek(file_, 16, SEEK_SET);
-	fread(&chunk_size, 1, 4, file_);
+bool WavDecoder::Open(Filesystem::InputStream stream) {
+	this->stream = stream;
+	stream->seekg(16, std::ios::ios_base::beg);
+	stream->read(reinterpret_cast<char*>(&chunk_size), sizeof(chunk_size));
 	Utils::SwapByteOrder(chunk_size);
-	fseek(file_, 2, SEEK_CUR);
-	fread(&nchannels, 1, 2, file_);
+	stream->seekg(2, std::ios::ios_base::cur);
+	stream->read(reinterpret_cast<char*>(&nchannels), sizeof(nchannels));
 	Utils::SwapByteOrder(nchannels);
-	fread(&samplerate, 1, 4, file_);
+	stream->read(reinterpret_cast<char*>(&samplerate), sizeof(samplerate));
 	Utils::SwapByteOrder(samplerate);
-	fseek(file_, 6, SEEK_CUR);
+	stream->seekg(6, std::ios::ios_base::cur);
 	uint16_t bitspersample;
-	fread(&bitspersample, 1, 2, file_);
+	stream->read(reinterpret_cast<char*>(&bitspersample), sizeof(bitspersample));
```

Comment:
This is a good case where doing a single read into a buffer and copying the pieces out would be faster than all these individual reads and seeks.